### PR TITLE
Fix for Issue #401

### LINKED ISSUE #401: References in tutorials
The references in tutorials are manual entries. It would be better to instead make use of the changes introduced in #228.

To close this issue: 

- [ ] use `articles.bib` or `books.bib` for the tutorials references
- [ ] use appropriate `bibtex` keys for inline citations in tutorials
- [ ] add a references section at the bottom of each tutorial  (files are in `docs`)

https://chiplicity.readthedocs.io/en/latest/Using_Sphinx/UsingBibTeXCitationsInSphinx.html

https://toqito.readthedocs.io/en/latest/tutorials.html

### PR DESCRIPTION
**Pull request by Roshan-Thomas**
## Description
References in the tutorials have been fixed as per the guidelines in #401. Fix is part of the Unitary Hack 2024.

Fixes #401 

## Changes
The changes are listed below.

- The references have automatically been updated in each tutorial using the Spinx tool.
- Some references in the tutorials were missing from the articles.bib file, so I have added them using the same style as recommended.

### CODE REVIEW

File: docs/articles.bib [hunk 1]
```diff
@@ -90,6 +90,19 @@ @article{Breuer_2006_Optimal
 
 
 #Last name begins with C
+@article{Cavalcanti_2015_Detection,
```
Comment (purva-thakre) [inline docs/articles.bib:93]: @Roshan-Thomas Thank you for this PR! 

Could you please make sure the keys are arranged alphabetically? You have placed them in the right spot but a key beginning with `cav` should be placed after `cab` and so on.

Comment (Roshan-Thomas) [reply docs/articles.bib]: Hi, I have fixed the issue in the following commit. Hope that fixes everything :)

**Review by purva-thakre — APPROVED** [review]
LGTM! Thanks @Roshan-Thomas. Feel free to comment on the linked issue so that the issue can be assigned to you for the bounty bot. 

@vprusso Do you want to take a look at this before we merge it?

**Comment by vprusso** [comment]
Looks good on my end as well! Thank you for the contribution, @Roshan-Thomas and thank you for the additional help in guiding the PR, @purva-thakre ! 

Feel free to merge when ready and congrats, @Roshan-Thomas !

**Comment by purva-thakre** [comment]
Congrats on the bounty @Roshan-Thomas !

**Pull request merged**